sdk/rust/oak/src/lib.rs

```diff
@@ -139,6 +162,10 @@ pub fn wait_on_channels(handles: &[ReadHandle]) -> Result<Vec<ChannelReadStatus>
 pub fn channel_read(half: ReadHandle, buf: &mut Vec<u8>, handles: &mut Vec<Handle>) -> OakStatus {
     // Try reading from the channel twice: first with provided vectors, then
     // with vectors that have been resized to meet size requirements.
+
+    // We cannot deserialize directly into the handles vector because `Handle` may
+    // not have the correct memory layout.
+    let mut handles_buf = vec![];
```

Comment:
Should this be `Vec::with_capacity(handles.capacity() * 8)`, as the caller may well have accurate expectations about how many handles to expect?

Reply:
Done